### PR TITLE
Merging ubuntu scripts

### PR DESCRIPTION
**Pull request by bam241**
This should address #37 

what do you think about it ? @gonuke 

@Mongwell would you have time to test this ?

usage would be:
`./ubuntu.sh dev` or `./ubuntu.sh stable` (the script will attempt to detect you version (and exist if failed to do so)
or
`./ubuntu.sh dev 18.04` or ./ubuntu.sh stable 16.04`...

### CODE REVIEW

**Comment by gonuke** [comment]
kind of like the dockerfile discussion - I worry that these undermine the ability for the most basic user to learn from the file about how to install our code...

Interested in other thoughts

**Comment by Mongwell** [comment]
I will definitely test this update on a few VMs (or just Ubuntu Docker containers) tomorrow. 

In regard to @gonuke's comment, while I agree that these changes add some complexity to this particular script file, I think the script file `ubuntu_mint.sh` contains all the important details on how to properly install PyNE. The files being merged here mainly serve the purpose of creating variables with the correct list of dependencies, which are well-documented for new users.

I do think it should be easy to determine the exact package names for all the dependencies, so maybe a few comments around those parts of the script to draw the attention of someone reading the file can handle that.

**Comment by gonuke** [comment]
What if we made bash functions for parts of this to add readability:
* `detect_version()`
* `validate_version(version)`
* `set_base_config()`
* `update_config(version)`

That might be overkill, but will make it more approachable?

**Comment by gonuke** [comment]
What if we made bash functions for parts of this to add readability:
* `detect_version()`
* `validate_version(version)`
* `set_base_config()`
* `update_config(version)`

That might be overkill, but will make it more approachable?

**Comment by bam241** [comment]
I think adding those function would add readability and probably maintainability... 

About the ability to learn how to install our software... I am not completely convinced that the scripts actually help new users to learn how to install `PyNE`. While I can see how the DockerFile helps them to get introduced to docker environment... 
I am doubting of their usefulness of both (script & docker) as installation learning tools...
I think an install documentation,line by line, helping them to learn by doing, is a better than a script, where users learn by reading...
I am not 100% sure of this, but I have the feeling that providing script for new users speed-up the short term efficiency but slow down the long term one...

That said I am probably bias, and "shadowed" by my own experience and knowledge....

File: ubuntu.sh [hunk 1]
```diff
@@ -5,53 +5,94 @@
...
+proccess_args $*
+# Check is the Ubuntu version is a supported one
+validate_version ${ubuntu_version}
+
+# Set the base config
+set_base_config
+# Update the configuration based on the ubuntu version
+update_config ${ubuntu_version}
+
+source Ubuntu_mint.sh $pyne_version
```
Comment (Mongwell) [inline ubuntu.sh]: Actual script filename is not capitalized. Should be `ubuntu_mint.sh`

**Comment by Mongwell** [comment]
If the first argument is not either `dev` or `stable`, should the script exit unsuccessfully? It isn't currently.

File: ubuntu.sh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+# After the build finishes run:
+#  $ source ~/.bashrc
+# or open a new terminal.
+
+# Use package manager for as many packages as possible
+
+
+# process the arguments
+function proccess_args() {
+    if [ $# -lt 2 ]
```
Comment (gonuke) [inline ubuntu.sh]: What if there are 0 arguments?

File: ubuntu.sh [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+function validate_version() {
+    local arg1=$1
+    if [[ $arg1 == "" ]]
+    then
+        echo "\
+Unable to determine the system Ubuntu version, using the default one: 18.04"    
+    elif [[ $arg1 != "16.04" ]] && [[ $arg1 != "18.04" ]]
+    then
+        echo "\
+Only Ubuntu 16.06 and 18.04 are supported by this script. Use at your own risk!"
```
Comment (gonuke) [inline ubuntu.sh]:
```suggestion
Only Ubuntu 16.04 and 18.04 are supported by this script.
Proceeding with configuration for 18.04.
Use at your own risk!"
```

**Comment by bam241** [comment]
@Mongwell @gonuke : your comments are related.
`ubuntu_mint.sh` script had a default (which is `dev`) as it only has a trigger on `stable`, so using `ubuntu.sh` without argument will trigger:
- ubuntu version detection
- `dev` build

we can change that if you want.. or make it more explicit...

**Comment by Mongwell** [comment]
I think those defaults are fine, but if the script is run as `./ubuntu.sh badarg` or `./ubuntu.sh 18.04`, the script will store the argument into the variable `pyne_version`, and run the version detection. My guess is that this will break the script, but only once it reaches the conditional in `ubuntu_mint.sh` that checks whether to checkout the stable branch of PyNE.

I think the solution would be to exit if the first argument exists (so is not equal to `"" `) but isn't either 'dev' or 'stable'.

**Comment by bam241** [comment]
fair point !

**Comment by Mongwell** [comment]
I tested the script on 4 clean Ubuntu VM's. The stable version builds fine on both 16 and 18. The dev version doesn't work currently, as it needs the `future` Python package. We could add that to the list, but switching over to Python 3 might be the better option.

**Comment by kkiesling** [comment]
not sure what this PR status is, but FYI I changed the base branch to be `main`.

**Comment by bam241** [comment]
to old an overly complicated... closing... :)

**Pull request closed without merging**